ixwebsocket/IXWebSocketTransport.cpp

```diff
@@ -176,33 +189,67 @@ namespace ix
         _onCloseCallback = onCloseCallback;
     }
 
-    // Only consider send time points for that computation.
-    // The receive time points is taken into account in Socket::poll (second parameter).
+    // Only consider send PING time points for that computation.
     bool WebSocketTransport::heartBeatPeriodExceeded()
     {
-        std::lock_guard<std::mutex> lock(_lastSendTimePointMutex);
+        std::lock_guard<std::mutex> lock(_lastSendPingTimePointMutex);
         auto now = std::chrono::steady_clock::now();
-        return now - _lastSendTimePoint > std::chrono::seconds(_heartBeatPeriod);
+        return now - _lastSendPingTimePoint > std::chrono::seconds(_heartBeatPeriod);
     }
 
-    void WebSocketTransport::poll()
+    bool WebSocketTransport::pongReceiveDelayExceeded()
     {
-        _socket->poll(
-            [this](PollResultType pollResult)
+        if (_heartBeatFactorDisconnectOnNoResponse == kDefaultHeartBeatFactorDisconnectOnNoResponse)
+            return false;
+
+        std::lock_guard<std::mutex> lock(_lastReceivePongTimePointMutex);
+        auto now = std::chrono::steady_clock::now();
+        return now - _lastReceivePongTimePoint > std::chrono::seconds(_heartBeatFactorDisconnectOnNoResponse*_heartBeatPeriod);
+    }
+
+    void WebSocketTransport::runHeartBeat()
```

Comment:
Better name it `heartbeatWatchdogThreadFunc`

Reply:
agree